### DISCUSSION TITLE
ckb 0.117.0 (4920ecf 2024-07-29)

### DISCUSSION THREAD

**Discussion by doitian** [nervosnetwork/ckb#4558] [Announcements]
### Highlights

There are two improvements applied for the RPC:

-   #4454: Add `include_tx_pool: Option<bool>` param to `ChainRpcImpl::get_live_cell`' (@eval-exec)
-   #4486: Add `assume_valid_target_reached: bool` to `NetRpc::sync_state` (@eval-exec)

Please note that these changes are considered breaking changes and may cause compatibility issues with older versions of RPC clients.

### Compatibility Table

| Network | Consensus Version        | Minimal CKB Version |
|---------|------------------------- | ------------------- |
| dev     | ckb2023 since epoch 0    |                     |
| testnet | ckb2021 since epoch 3113 | v0.101.0            |
| mainnet | ckb2021 since epoch 5414 | v0.103.0            |

See [how to connect to various chains](https://github.com/nervosnetwork/ckb/wiki/Chains) in Wiki.

### Default assume valid target

- Mainnet: [HASH](https://explorer.nervos.org/block/HASH)
- Testnet: [HASH](https://explorer.nervos.org/block/HASH)

### Downloads

| OS                | Arch  | Runtime Dependencies | Package                 | Sign                |
| ----------------- | ----- | -------------------- | ----------------------- | ------------------- |
| macOS             | x64   | macOS 10.15 or above | [zip][macos-zip]        | [PGP][macos-pgp]    |
| macOS  (Portable) | x64   | macOS 10.15 or above | [zip][macos-p-zip]      | [PGP][macos-p-pgp]  |
| macOS             | ARM64 | macOS 10.15 or above | [zip][macosa-zip]       | [PGP][macosa-pgp]   |
| macOS  (Portable) | ARM64 | macOS 10.15 or above | [zip][macosa-p-zip]     | [PGP][macosa-p-pgp] |
| Linux             | x64   | glibc, libstdc++     | [tarball][linux-zip]    | [PGP][linux-pgp]    |
| Linux  (Portable) | x64   | glibc, libstdc++     | [tarball][linux-p-zip]  | [PGP][linux-p-pgp]  |
| CentOS            | x64   | glibc, libstdc++     | [tarball][centos-zip]   | [PGP][centos-pgp]   |
| CentOS (Portable) | x64   | glibc, libstdc++     | [tarball][centos-p-zip] | [PGP][centos-p-pgp] |
| Windows           | x64   | VC++ Redistributable | [zip][windows-zip]      | [PGP][windows-pgp]  |
| Linux             | ARM64 | glibc, libstdc++     | [tarball][arm-zip]      | [PGP][arm-pgp]      |
| Docker            |       |                      | [dockerhub][]           |                     |

The `.asc` files are signatures. It is wise and more secure to check out for the files [integrity](https://github.com/nervosnetwork/ckb/blob/develop/docs/integrity-check.md).

Try the portable variants if the binaries complains about illegal instructions.

[macos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-apple-darwin.zip
[macos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-apple-darwin.zip.asc
[macos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-apple-darwin-portable.zip
[macos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-apple-darwin-portable.zip.asc
[macosa-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_aarch64-apple-darwin.zip
[macosa-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_aarch64-apple-darwin.zip.asc
[macosa-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_aarch64-apple-darwin-portable.zip
[macosa-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_aarch64-apple-darwin-portable.zip.asc
[linux-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-unknown-linux-gnu.tar.gz
[linux-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-unknown-linux-gnu.tar.gz.asc
[linux-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-unknown-linux-gnu-portable.tar.gz
[linux-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-unknown-linux-gnu-portable.tar.gz.asc
[centos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-unknown-centos-gnu.tar.gz
[centos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-unknown-centos-gnu.tar.gz.asc
[centos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-unknown-centos-gnu-portable.tar.gz
[centos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-unknown-centos-gnu-portable.tar.gz.asc
[windows-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-pc-windows-msvc.zip
[windows-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_x86_64-pc-windows-msvc.zip.asc
[arm-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_aarch64-unknown-linux-gnu.tar.gz
[arm-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.117.0/ckb_v0.117.0_aarch64-unknown-linux-gnu.tar.gz.asc
[dockerhub]: https://hub.docker.com/r/nervos/ckb/tags

#### Features

-   #4454: Add `include_tx_pool: Option<bool>` param to `ChainRpcImpl::get_live_cell`' (@eval-exec)

    This is a breaking change to the RPC.

-   #4486: Add `assume_valid_target_reached: bool` to `NetRpc::sync_state` (@eval-exec)

    This is a breaking change to the RPC.

#### Bug Fixes

-   #4484: Fix rich indexer `partial` query by args performance issue (@EthanYuan)
-   #4505: Fix websocket subscription performance issue (@chenyukang)

#### Improvements

-   #4487: tweak `max_ancestors_count` (@zhangsoledad)
-   #4459: Use standalone runtime for RPC service (@chenyukang)
-   #4511: Modify the record scope of tx-pool reject record and fix rule for orphan tx. (@zhangsoledad)
-   #4531: Early return `process_fetch_cmd` if ckb received exit signal (@eval-exec)

<hr /><em>This discussion was created from the release <a href='https://github.com/nervosnetwork/ckb/releases/tag/v0.117.0'>ckb 0.117.0 (4920ecf 2024-07-29)</a>.</em>

**Comment by joshyates1980** [comment]
I use ckb run --indexer command. I just upgraded to ver 0.117.  Does the Indexer have to be rebuilt after each upgrade:  INFO ckb_indexer_sync append 000000 to xxxxxxx?  Also, is there a particular reason why CKB Node Probe cannot be included in upgrades?  I manually edit the files (which is fine): https://nodes.ckb.dev/getConnectedInstruction

**Reply by eval-exec** [reply]
> I use ckb run --indexer command. I just upgraded to ver 0.117. Does the Indexer have to be rebuilt after each upgrade: INFO ckb_indexer_sync append 000000 to xxxxxxx? Also, is there a particular reason why CKB Node Probe cannot be included in upgrades? I manually edit the files (which is fine): https://nodes.ckb.dev/getConnectedInstruction


Could you please provide your indexer related configuration section from the `ckb.toml` file?

**Comment by quake** [comment]
> I use ckb run --indexer command. I just upgraded to ver 0.117. Does the Indexer have to be rebuilt after each upgrade: INFO ckb_indexer_sync append 000000 to xxxxxxx? 

Re-indexing is only needed if the indexer storage schema changes, what was the version of ckb before you upgraded?

**Reply by joshyates1980** [reply]
ver 0.116.1

**Reply by quake** [reply]
there are no data migration between 0.116.1 and 0.117.0, we cannot reproduce this bug.
